logging/google/cloud/logging/handlers/_helpers.py

```diff
@@ -69,6 +79,33 @@ def get_trace_id_from_flask():
     return trace_id
 
 
+def get_trace_id_from_webapp2():
+    """Get trace_id from webapp2 request headers.
+
+    :rtype: str
+    :return: Trace_id in HTTP request headers.
+    """
+    if webapp2 is None:
+        return None
+
+    try:
+        # get_request() succeeds if we're in the middle of a webapp2
+        # request, or raises an assertion error otherwise:
+        # "Request global variable is not set".
```

Comment:
Yeah, the AssertionError thing isn't part of the API :-), and I'm kinda worried that one day webapp2 will get updated in a way that raises a different exception.  But I think the tests will notice if that ever happens.